chain/emitter/handler.go

```diff
@@ -2,24 +2,53 @@ package emitter
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/tendermint/abci/types"
 
 	"github.com/bandprotocol/bandchain/chain/x/oracle"
+	"github.com/bandprotocol/bandchain/chain/x/oracle/types"
 )
 
-// handleMsg handles the given message by publishing relevant events and populates accounts
-// that need balance update in 'app.accs'. Also fills in extra info for this message.
-func (app *App) handleMsg(txHash []byte, msg sdk.Msg, log sdk.ABCIMessageLog, extra JsDict) {
+func parseEvents(events sdk.StringEvents) EvMap {
 	evMap := make(EvMap)
-	for _, event := range log.Events {
+	for _, event := range events {
 		for _, kv := range event.Attributes {
 			key := event.Type + "." + kv.Key
 			evMap[key] = append(evMap[key], kv.Value)
 		}
 	}
+	return evMap
+}
+
+// handleMsg handles the given message by publishing relevant events and populates accounts
+// that need balance update in 'app.accs'. Also fills in extra info for this message.
+func (app *App) handleMsg(txHash []byte, msg sdk.Msg, log sdk.ABCIMessageLog, extra JsDict) {
+	evMap := parseEvents(log.Events)
 	switch msg := msg.(type) {
 	case oracle.MsgRequestData:
 		app.handleMsgRequestData(txHash, msg, evMap, extra)
 	case oracle.MsgReportData:
 		app.handleMsgReportData(txHash, msg, evMap, extra)
 	}
 }
+
+func (app *App) handleEndBlock(event abci.Event) {
+	events := sdk.StringifyEvents([]abci.Event{event})
+	evMap := parseEvents(events)
+	switch event.Type {
+	case types.EventTypeRequestExecute:
+		resolveStatus := types.ResolveStatus(atoi(evMap[types.EventTypeRequestExecute+"."+types.AttributeKeyResolveStatus][0]))
+		dict := JsDict{
+			"id":             atoi(evMap[types.EventTypeRequestExecute+"."+types.AttributeKeyRequestID][0]),
+			"request_time":   atoi(evMap[types.EventTypeRequestExecute+"."+types.AttributeKeyRequestTime][0]),
```

Comment:
Let's get all information apart from `id` from calling into the keeper's `GetResult` function. I think it's safer.

Reply:
This is not resolved